monitoring/mongodb/dashboard.py

```diff
@@ -409,59 +725,75 @@ def mongodb_state_timeline(title, expr, description="", mappings=None, **kwargs)
 # Status Panels
 smallest_oplog = mongodb_stat(
     "Smallest oplog window",
-    expr=MongoDBMetrics.oplog_window(),
+    expr="min("
+    + Metrics.OPLOG_HEAD_TIMESTAMP()
+    + " - "
+    + Metrics.OPLOG_TAIL_TIMESTAMP()
+    + ")",
     format=UNITS.SECONDS,
     thresholds=THRESHOLDS["oplog_window"],
     description="This number represents the history window of the currently most used MongoDB shard. Losing a secondary for more than this duration will result in a divergence.",
 )
 
-
 disk_space_util = mongodb_stat(
     "Disk Space Utilization",
-    expr=MongoDBMetrics.disk_usage(),
+    expr="avg("
+    + Metrics.DBSTATS_FS_USED_SIZE()
+    + ") / avg("
+    + Metrics.DBSTATS_FS_TOTAL_SIZE()
+    + ")",
     format=UNITS.PERCENT_UNIT,
     thresholds=THRESHOLDS["disk_usage"],
     minValue=0,
     maxValue=100,
     description="Shows information about the disk space usage of the specified mountpoint.",
 )
 
-
 num_indexes = mongodb_stat(
     "Number of indexes",
     targets=[
         Target(
-            expr=f'sum(mongodb_dbstats_indexes{{{DATABASE_FILTER_USER}, rs_state="{RS_STATE_PRIMARY}", job=~"$job", namespace="{NAMESPACE}"}})',
+            expr="sum("
+            + Metrics.DBSTATS_INDEXES(
+                DATABASE_FILTER_USER, f'rs_state="{RS_STATE_PRIMARY}"'
+            )
+            + ")",
             legendFormat="Metadata database",
         )
     ],
 )
 
-
 num_collections = mongodb_stat(
     "Number of Collections",
     targets=[
         Target(
-            expr=f'sum(mongodb_mongod_db_collections_total{{{DATABASE_FILTER_USER}, rs_state="{RS_STATE_PRIMARY}", job=~"$job", namespace="{NAMESPACE}"}})',
+            expr="sum("
+            + Metrics.DB_COLLECTIONS_TOTAL(
+                DATABASE_FILTER_USER, f'rs_state="{RS_STATE_PRIMARY}"'
+            )
+            + ")",
             format="table",
             instant=True,
             legendFormat="__auto",
         )
     ],
 )
 
-
 # State Timeline Panels
 shard_server_states = mongodb_state_timeline(
     "Shard Server States",
-    MongoDBMetrics.replset_state("zenko/data-db-mongodb-sharded-shard.*-data"),
+    "max by (pod) ("
+    + Metrics.REPLSET_STATE('job=~"zenko/data-db-mongodb-sharded-shard.*-data"')
```

Comment:
job filter(s) should ideally not be hard-coded, but come from ZKOP...
what we often do is that ZKOP sets multiple params, for each kind of job: so in this case it may be job_configsvr, job_shards, ...

Though in this instance, Mongo is deployed by the chart in this repo, so there will always be some coupling :-/ ... So at the very least it should be made consistent:

```suggestion
    + Metrics.REPLSET_STATE('job=~"${namespace}/${job}-shard.*-data"')
```